src/main/scala/is/hail/expr/ir/AggOp.scala

```diff
@@ -0,0 +1,109 @@
+package is.hail.expr.ir
+
+import is.hail.annotations.aggregators._
+import is.hail.asm4s._
+import is.hail.expr._
+import is.hail.utils._
+
+sealed trait NullaryAggOp { }
+final case class Fraction() extends NullaryAggOp { } // remove when prefixes work
```

Comment:
I am referring to the optimization that takes a pair of aggregation expressions:
```
aggregable.agg1()
aggregable.agg2()
```
and emits them as:
```
emit(aggregable)
DUP
emit(agg1)
emit(agg2)
```

If that exists,

```
aggregable.fraction()
```

can be desugared to:

```
aggregable.sum() / aggregable.count()
```

NB: I already plan to not define Fraction for any types other than Boolean, because we can always desugar `agg.fraction(p)` on an arbitrary type to `agg.map(p).fraction()`.